src/LongRunning/Gapic/OperationsGapicClient.php

```diff
@@ -226,32 +216,23 @@ public function __construct($options = [])
             $this->descriptors[$method]['pageStreamingDescriptor'] = $pageStreamingDescriptor;
         }
 
+        $this->transport = $this->determineTransport($options);
+
         $clientConfigJsonString = file_get_contents(__DIR__.'/../resources/operations_client_config.json');
         $clientConfig = json_decode($clientConfigJsonString, true);
         $this->defaultCallSettings =
                 CallSettings::load(
                     'google.longrunning.Operations',
                     $clientConfig,
                     $options['retryingOverride'],
-                    GrpcConstants::getStatusCodeNames(),
+                    $this->transport->getStatusCodeNames(),
                     $options['timeoutMillis']
                 );
 
         $this->scopes = $options['scopes'];
 
-        $createStubOptions = [];
-        if (array_key_exists('sslCreds', $options)) {
-            $createStubOptions['sslCreds'] = $options['sslCreds'];
-        }
-        $this->grpcCredentialsHelper = new GrpcCredentialsHelper($options);
-
-        $createOperationsStubFunction = function ($hostname, $opts, $channel) {
-            return new OperationsGrpcClient($hostname, $opts, $channel);
-        };
-        if (array_key_exists('createOperationsStubFunction', $options)) {
-            $createOperationsStubFunction = $options['createOperationsStubFunction'];
-        }
-        $this->operationsStub = $this->grpcCredentialsHelper->createStub($createOperationsStubFunction);
+        $stubClassName = sprintf('Google\LongRunning\Operations%sClient', ucfirst($this->transport->getName()));
+        $this->operationsStub = $this->transport->createStub($stubClassName, $options);
```

Comment:
This method should probably take something other than a class name, although I don't know what it would be.